language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -209,18 +199,28 @@ genModule pkgMap (Scope scope) curPkgId mod
         imports = (PRSelf, modName) `Set.delete` Set.unions refs
         (internalImports, externalImports) = splitImports imports
         rootPath = map (const "..") (unModuleName modName)
-        modText = T.unlines $ intercalate [""] $ filter (not . null) $
-          modHeader
-          : map (externalImportDecl pkgMap) (Set.toList externalImports)
-          : map (internalImportDecl rootPath) internalImports
-          : map (map renderTsDecl) decls
+        makeMod jsSyntax body = T.unlines $ intercalate [""] $ filter (not . null) $
+          modHeader jsSyntax
+          : map (externalImportDecl jsSyntax pkgMap) (Set.toList externalImports)
+          : map (internalImportDecl jsSyntax rootPath) internalImports
+          : body
+
+        (jsBody, tsDeclsBody) = unzip $ map (unzip . map renderTsDecl) decls
         depends = Set.map (Dependency . pkgRefStr pkgMap) externalImports
-   in Just (modText, depends)
+   in Just ((makeMod ES5 jsBody, makeMod ES6 tsDeclsBody), depends)
   where
     modName = moduleName mod
     tpls = moduleTemplates mod
     serDefs = defDataTypes mod
-    modHeader =
+    modHeader ES5 = commonjsPrefix ++
+      [ "/* eslint-disable-next-line no-unused-vars */"
+      , "var jtv = require('@mojotech/json-type-validation');"
```

Comment:
I just learnt that `const` was only introduced in ES6. 😞

Reply:
Yeah all the good stuff is only in ES6 :cry: But given that this is generated code it luckily isn’t a big deal (and I believe the typescript typechecker might still yell at you although I haven’t verified that (should be the same as before)).